libs/media-player/src/lib/audio-clip-player.tsx

```diff
@@ -0,0 +1,101 @@
+import { isNullOrUndefined } from '@coscrad/validation-constraints';
+import BugReportIcon from '@mui/icons-material/BugReport';
+import LinkOffIcon from '@mui/icons-material/LinkOff';
+import { Box, CircularProgress, IconButton, Tooltip } from '@mui/material';
+import { useEffect, useState } from 'react';
+import { DefaultPlayButton } from './default-clip-player-button';
+
+interface IPlayButtonProps {
+    onButtonClick: () => void;
+}
+
+export interface IPlayButton {
+    (props: IPlayButtonProps): JSX.Element;
+}
+
+export interface AudioClipPlayerProps {
+    audioUrl: string;
+    UserDefinedPlayButton?: IPlayButton;
+}
+
+enum AudioState {
+    loading = 'loading',
+    error = 'error',
+    canPlay = 'canPlay',
+    undefined = 'undefined',
+}
+
+export function AudioClipPlayer({ audioUrl, UserDefinedPlayButton }: AudioClipPlayerProps) {
+    const [audioState, setAudioState] = useState<AudioState>(AudioState.loading);
+
+    useEffect(() => {
+        const onCanPlay = () => {
+            setAudioState(AudioState.canPlay);
+        };
+
+        const onError = () => {
+            setAudioState(AudioState.error);
+        };
+
+        /**
+         * This audio instance will never be played. We are using it to test whether
+         * the URL is valid.
+         */
+        const testAudio = new Audio(audioUrl);
+
+        testAudio.addEventListener('canplaythrough', onCanPlay);
+
+        testAudio.addEventListener('error', onError);
+
+        if (!isNullOrUndefined(audioUrl)) {
```

Comment:
Move the null check to the top of the effect and return early in that case. Remove the redundant test audio instance and event handler. I.e., delete lines 50-56 and above line 32 add
```ts
if(isNullOrUndefined(audioUrl)) setAudioState(AudioState.undefined);
```

Reply:
👍